### PR TITLE
Make MAX_CONCURRENT_BUILDS configurable using the command line

### PR DESCRIPTION
**Pull request by andreafioraldi**
This should solve #946 .
An user with a limited hardware setup must use a low number of concurrent builds that now can be specified in the command line when starting the experiment.

E.g.:

```
PYTHONPATH=. python3 experiment/run_experiment.py \
  --experiment-config experiment-config.yaml \
  --benchmarks freetype2-2017 bloaty_fuzz_target \
  --experiment-name $EXPERIMENT_NAME \
  --fuzzers afl libfuzzer
  --concurrent-builds 1
``` 

The default is still 30, it will likely fail on any local experiment setup.

cc @xcoldhandsx

### CODE REVIEW

File: experiment/build/builder.py [hunk 1]
```diff
@@ -89,13 +89,16 @@ def build_measurer(benchmark: str) -> bool:
         return False
 
 
-def build_all_measurers(benchmarks: List[str]) -> List[str]:
+def build_all_measurers(
+        benchmarks: List[str],
+        concurrent_builds: int = DEFAULT_MAX_CONCURRENT_BUILDS) -> List[str]:
```
Comment (jonathanmetzman) [inline experiment/build/builder.py:94]: nit: Maybe "num_concurrent_builds."

File: experiment/build/builder.py [hunk 2]
```diff
@@ -118,12 +121,14 @@ def split_successes_and_failures(inputs: List,
     return successes, failures
 
 
-def retry_build_loop(build_func: Callable, inputs: List[Tuple]) -> List:
+def retry_build_loop(build_func: Callable, inputs: List[Tuple],
+                     concurrent_builds: int) -> List:
```
Comment (jonathanmetzman) [inline experiment/build/builder.py:125]: nit: "num_concurrent_builds."

File: experiment/build/builder.py [hunk 3]
```diff
@@ -118,12 +121,14 @@ def split_successes_and_failures(inputs: List,
...
     """Calls |build_func| in parallel on |inputs|. Repeat on failures up to
     |NUM_BUILD_RETRIES| times. Returns the list of inputs that |build_func| was
     called successfully on."""
     successes = []
-    with mp_pool.ThreadPool(MAX_CONCURRENT_BUILDS) as pool:
+    logs.info('Concurrent builds: %d', concurrent_builds)
```
Comment (jonathanmetzman) [inline experiment/build/builder.py:130]: nit: End the log message with a period.

File: experiment/build/builder.py [hunk 4]
```diff
@@ -159,8 +164,10 @@ def build_fuzzer_benchmark(fuzzer: str, benchmark: str) -> bool:
     return True
 
 
-def build_all_fuzzer_benchmarks(fuzzers: List[str],
-                                benchmarks: List[str]) -> List[str]:
+def build_all_fuzzer_benchmarks(
+        fuzzers: List[str],
+        benchmarks: List[str],
+        concurrent_builds: int = DEFAULT_MAX_CONCURRENT_BUILDS) -> List[str]:
```
Comment (jonathanmetzman) [inline experiment/build/builder.py:170]: nit: num_concurrent_builds.

File: experiment/build/builder.py [hunk 5]
```diff
@@ -192,10 +200,19 @@ def main():
                         help='Fuzzer names.',
                         nargs='+',
                         required=True)
+
+    parser.add_argument('-cb',
+                        '--concurrent-builds',
```
Comment (jonathanmetzman) [inline experiment/build/builder.py:205]: num_concurrent_builds.

**Review by jonathanmetzman — COMMENTED** [review]
Thanks a lot!
Just have some dumb nits to fix if you can. If not, I can merge anyway.

**Comment by andreafioraldi** [comment]
@jonathanmetzman any other things left?

**Pull request merged**